mesonbuild/interpreter.py

```diff
@@ -2229,7 +2252,13 @@ def do_subproject(self, dirname, kwargs):
             incpath = ' => '.join(fullstack)
             raise InvalidCode('Recursive include of subprojects: %s.' % incpath)
         if dirname in self.subprojects:
-            return self.subprojects[dirname]
+            subproject = self.subprojects[dirname]
+
+            if required and not subproject.found():
+                raise InterpreterException('Subproject "%s/%s" required but not found.' % (
+                        self.subproject_dir, dirname))
```

Comment:
__[Flake8]__

[E126] continuation line over-indented for hanging indent


<!-- Comment by [Sider](https://sider.review) -->